File: Server/PowerCraft/Tools/JSON.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Newtonsoft.Json
Imports System.IO

Namespace PowerCraft.Tools
    Public Class JSONWriter
        Public Shared Sub SaveConfig([cObj] As ConfigObj)
            Dim text As [String] = JsonConvert.SerializeObject([cObj])
            Dim textWriter As New Writer(text)
            textWriter.WriteToTextFile("config.json")
        End Sub
    End Class

    Public Class JSONReader
        Public Shared Function [cObj]() As ConfigObj
            Dim json As String = File.ReadAllText("config.json")
            Return JsonConvert.DeserializeObject(Of ConfigObj)(json)
        End Function
    End Class
End Namespace

File: PowerCraft/Tools/JSON.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Newtonsoft.Json;
using System.IO;

namespace PowerCraft.Tools
{
    public class JSONWriter
    {
        public static void SaveConfig(ConfigObj cObj)
        {
            String text = JsonConvert.SerializeObject(cObj);
            Writer textWriter = new Writer(text);
            textWriter.WriteToTextFile("config.json");
        }
    }

    public class JSONReader
    {
        public static ConfigObj cObj()
        {
            string json = File.ReadAllText(@"config.json");
            return JsonConvert.DeserializeObject<ConfigObj>(json);
        }
    }
}
